routes/artifacts.js

```diff
@@ -296,17 +308,30 @@ api.declare({
         bucket = this.privateBucket;
       }
       // Create put URL
+      var metadata = {};
+      if (artifact.details.contentSHA256) {
+        metadata['content-sha256'] = artifact.details.contentSHA256;
+      }
       var putUrl = await bucket.createPutUrl(
         artifact.details.prefix, {
         contentType:      artifact.contentType,
-        expires:          30 * 60 + 10 // Add 10 sec for clock drift
+        expires:          30 * 60 + 10, // Add 10 sec for clock drift
+        contentMD5:       artifact.details.contentMD5 || undefined,
```

Comment:
Wouldn't this either be defined or undefined?  Or are you doing this in case contentMD5 is an empty string?